src/test/java/org/opensearch/security/dlic/rest/api/SavedTenantApiTest.java

```diff
@@ -0,0 +1,134 @@
+
+        // add user
+        addUserWithPassword(testUser, testPass, HttpStatus.SC_CREATED);
+
+        // test - unauthorized access as credentials are missing.
+        HttpResponse response = rh.executeGetRequest(savedTenantEndpoint);
+        assertEquals(HttpStatus.SC_UNAUTHORIZED, response.getStatusCode());
+
+        // test - incorrect password
+        response = rh.executeGetRequest(savedTenantEndpoint, encodeBasicHeader(testUser, testPass + "invalidating text"));
+        assertEquals(HttpStatus.SC_UNAUTHORIZED, response.getStatusCode());
+
+        // test - incorrect user; not possible to check until later (when adding a (target) user parameter in body)
+        // can't do this until user manager; check if only users who can manage this user can get/set this info
+
+        // test - valid request
+        response = rh.executeGetRequest(savedTenantEndpoint, encodeBasicHeader(testUser, testPass));
+        assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+        Settings body = Settings.builder().loadFromSource(response.getBody(), XContentType.JSON).build();
+        assertEquals(defaultTenantValue, body.get("saved_tenant"));
+    }
+
+    @Test
+    public void testSet_tenant() throws Exception{
+        // arrange
+        setup();
+        final String testUser = "test-user";
+        final String testPass = "test-pass";
+        final String defaultTenantValue = "global-tenant";
+        final String newSavedTenant = "shinjuku";
+        final String changeSavedTenantPayload = "{\"saved_tenant\":\"" + newSavedTenant + "\"}";
+        final String savedTenantEndpoint = BASE_ENDPOINT + "account/saved_tenant";
+
+        // add user
+        addUserWithPassword(testUser, testPass, HttpStatus.SC_CREATED);
+
+        // check newly created user has valid saved_tenant value
+        HttpResponse response = rh.executeGetRequest(savedTenantEndpoint, encodeBasicHeader(testUser, testPass));
+        assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+        Settings body = Settings.builder().loadFromSource(response.getBody(), XContentType.JSON).build();
+        assertEquals(defaultTenantValue, body.get("saved_tenant"));
+        
+        // test - unauthorized access as credentials are missing.
+        response = rh.executePutRequest(savedTenantEndpoint, changeSavedTenantPayload);
+        assertEquals(HttpStatus.SC_UNAUTHORIZED, response.getStatusCode());
+
+        // test - incorrect password
+        response = rh.executePutRequest(savedTenantEndpoint, changeSavedTenantPayload, encodeBasicHeader(testUser, testPass + "invalidating text"));
+        assertEquals(HttpStatus.SC_UNAUTHORIZED, response.getStatusCode());
+
+        // test - incorrect user; not possible to check until later (when adding a (target) user parameter in body)
+        // can't do this until user manager; check if only users who can manage this user can get/set this info
+
+        // test - specified tenant does not exist
```

Comment:
If this is something to do, please use `TODO:` at the very beginning of the line of the text.